chain/consensus/hierarchical/gen/gen.go

```diff
@@ -10,6 +10,7 @@ func main() {
 	if err := gen.WriteTupleEncodersToFile("./cbor_gen.go", "hierarchical",
 		hierarchical.ConsensusParams{},
 		hierarchical.Validator{},
+		hierarchical.MiningParams{},
```

Comment:
Why do you need `MiningParams` to be CBOR-serializable? I don't think you are using it anywhere where this struct needs to implement `CBORMarshaler` (or maybe I've missed it).

Reply:
Yes, I thought we have to serialized all API input

Reply:
I don't think you need this. Try removing this, re-generate the cbor-gen and see if it works. IIRC there's no need for this if you are only going to use it as an API argument.